org.eclipse.xtext.xbase.tests/src/org/eclipse/xtext/xbase/tests/validation/XbaseValidationTest.xtend

```diff
@@ -1128,4 +1128,12 @@ class XbaseValidationTest extends AbstractXbaseTestCase {
 			}
 		'''.expression.assertNoErrors
 	}
+
+	@Test def void testInvalidNestedReturnBug406762() {
+		'''
+			{
+				return return 0
```

Comment:
Right! In all of the above cases we should get a validation error, right? Or are there situations when a nested return is considered as valid?